features/Dockerfile

```diff
@@ -0,0 +1,74 @@
+# syntax = docker/dockerfile:1.5
+# Used only for running tests using tox, see ../tox.ini
+ARG PG_MAJOR
+ARG PGHOME=/home/postgres
+ARG PGDATA=$PGHOME/data
+ARG LC_ALL=C.UTF-8
+ARG LANG=C.UTF-8
+
+FROM postgres:${PG_MAJOR}
+
+ARG PGHOME
+ARG PGDATA
+ARG LC_ALL
+ARG LANG
+
+ENV ETCDVERSION=3.3.13
+
+USER root
+RUN set -ex \
+    && apt-get update \
+    && apt-get reinstall init-system-helpers \
+    && apt-get install -y \
+      python3-pip \
+      python3-dev \
+      rsync \
+      curl \
+      gcc \
+      golang \
+      jq  \
+      locales  \
+      sudo \
+      busybox \
+      net-tools \
+      iputils-ping \
+    && rm -rf /var/cache/apt \
+    && python3 -m pip install --no-cache-dir tox \
+    \
+    && mkdir -p "$PGHOME" \
+    && sed -i "s|/var/lib/postgresql.*|$PGHOME:/bin/bash|" /etc/passwd \
+    && chown -R postgres:postgres /var/log /home/postgres \
+    \
+    # Download etcd \
+    && ETCDURL="https://github.com/coreos/etcd/releases/download/v$ETCDVERSION" \
+    && curl -sL "$ETCDURL/etcd-v$ETCDVERSION-linux-$(dpkg --print-architecture).tar.gz" \
+       | tar xz -C /usr/local/bin --strip=1 --wildcards --no-anchored etcd etcdctl
+
+
+# This Dockerfile syntax only works with docker buildx and the syntax
+# line at the top of this file.
+COPY <<EOF /tox-wrapper.sh
+#!/usr/bin/env bash
+set -ex
+copy_output() {
+    if [[ -d "$PGHOME/src/features/output" && /src/features ]] ;then
+        cp -a "$PGHOME/src/features/output" "/src/features/output-\$HOSTNAME"
+        find "/src/features/output-\$HOSTNAME" -type f -exec chmod 666 {} \\;
+        find "/src/features/output-\$HOSTNAME" -type d -exec chmod 777 {} \\;
+    fi
+}
+trap 'copy_output' SIGTERM
+# For architectures such as aarch we need to get the respective GOARCH
+# so we can tell etcd we're ok with running an unsupported architecture.
+export ETCD_UNSUPPORTED_ARCH=$(go env GOARCH)
+runuser -u postgres -- rsync -a --exclude=.tox /src/ "$PGHOME/src/"
+cd "$PGHOME/src"
+runuser -u postgres -w ETCD_UNSUPPORTED_ARCH -- "\$@" &
+wait $?
+copy_output
```

Comment:
I did think that originally yes, I'm not sure what it was that I read that made that problematic.

Reply:
No, the problem is that if you send an interrupt it's not seen by the shell that has the trap. It doesn't work with

```
trap copy_output SIGTERM SIGINT EXIT
runuser ...
```
or
```
trap copy_output SIGTERM
runuser ...
copy_output
```

Just
```
trap copy_output SIGTERM
runuser ... &
wait $!
copy_output
```

If it has this, it runs the copy twice on a successful run.
```
trap copy_output SIGTERM EXIT
runuser ... &
wait $?
copy_output
```

Finally, we need the sigterm trap in case the container is shutdown with `docker stop` or `docker kill`.